### ISSUE TITLE
No SuperCustomProperties for Tiled objects

### ISSUE THREAD

**Issue by Dalqin** [Seanba/SuperTiled2Unity#87]
I have a Tiled map with some objects in an object layer, using a ObjectType to map a Tiled object to a treasure chest prefab when importing the `.tsx` map file into Unity. In Tiled, this object has custom properties. I was hoping to see a `SuperCustomProperties` component with these custom properties upon importing my `.tsx` map into Unity, but I do not see a `SuperCustomProperties` component on my imported object.

Tiled - showing custom properties on an object
![image](https://user-images.githubusercontent.com/42662231/61985747-29272380-afd0-11e9-92de-74dcb2a9a5db.png)
Unity - hoping to see `SuperCustomProperties` with my "ItemData" property.
![image](https://user-images.githubusercontent.com/42662231/61985886-1cef9600-afd1-11e9-9e2d-98b678209fe4.png)

Looks like the Tiled object, my treasure chest, is being replaced with an GameObject identical to my treasure chest prefab (but not an instance of my prefab) with no custom properties that I can find. Am I looking in the right place?

I've two questions, assuming I have not made a mistake in setting this up:
1) Is there a way to get custom props on objects imported into Unity without any custom import script? My goal is to have a prefab treasure chest and use a custom property to define what item the treasure chest should reward the player when opened.
2) I found it interesting that the imported object is a replica of my prefab, but not an instance of the prefab. Was this an intended design decision or just how things came to be?

Thank you!

**Comment by Seanba** [comment]
Hi there, @Dalqin.

I wasn't aware that prefab instances were not being instantiated/referenced correctly. Can you check out the recently updated version 1.5.3 of SuperTiled2Unity? I'm now using a different method for instantiating prefabs and it appears to be working for me as expected.

As far as `Prefab Replacements` go and their relationship with `Custom Properties` this is something I handle through setting properties, fields, or methods on the prefab that is replacing the Tiled Object.

See this repository, and in particular this part of the documentation: https://github.com/Seanba/SuperPrefabReplacement#custom-properties-supported

**Comment by Dalqin** [comment]
I checked out 1.5.3 of SuperTiled2Unity and see my treasure chest Tiled object being imported as an instance of my treasure chest prefab in Unity. Yay!

The documentation you linked shows me I was expecting incorrect behavior regarding objects and custom properties. I was able to get my `ItemData` property value in Unity using a same-named public property, so thank you for pointing me in the right direction there. (And sorry for not reading all the documentation first!)

I dub my issue solved, and will close this ticket.
Thanks very much for the quick reply!
